File: pdaily-server-postgresql/src/main/java/co/icesi/pdaily/business/structure/patient/app/rest/PatientResource.java
package co.icesi.pdaily.business.structure.patient.app.rest;

import java.util.List;

import javax.inject.Inject;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;

import co.icesi.pdaily.business.structure.patient.app.PatientAppService;
import co.icesi.pdaily.business.structure.patient.app.PatientDTO;

@Path("/business/patient")
@Consumes(MediaType.APPLICATION_JSON)
@Produces(MediaType.APPLICATION_JSON)
public class PatientResource {
	@Inject
	PatientAppService appService;

	@POST
	public PatientDTO savePatient(PatientDTO dto) {
		return appService.savePatient( dto );
	}

	@GET
	public List<PatientDTO> findByPatient() {
		return appService.findAll();
	}

	@GET
	@Path("/{id}")
	public PatientDTO findById(@PathParam("id") String id) {
		return appService.findOrFail( id );
	}

	@DELETE
	@Path("/{id}")
	public void deletePatient(@PathParam("id") String id) {
		appService.deletePatient( id );
	}
}

File: backend/backend/src/main/kotlin/co/pdaily/cloud/patients/boundary/PatientResource.kt
package co.pdaily.cloud.patients.boundary

import co.pdaily.cloud.patients.entity.PatientDTO
import co.pdaily.cloud.patients.entity.toDTO
import java.util.*
import javax.inject.Inject
import javax.ws.rs.*
import javax.ws.rs.core.MediaType

@Path("/patients")
@Consumes(MediaType.APPLICATION_JSON)
@Produces(MediaType.APPLICATION_JSON)
class PatientResource {
    @Inject
    lateinit var patientService: PatientService

    @POST
    fun saveOrUpdate(patient: PatientDTO): PatientDTO {
        val saved = patientService.saveOrUpdate(patient.toPatient())
        return saved.toDTO()
    }

    @GET
    fun findAll(): List<PatientDTO> {
        val all = patientService.findAll()
        return all.map { it.toDTO() }
    }

    @GET
    @Path("/{id}")
    fun findById(@PathParam("id") id: UUID): PatientDTO {
        val found = patientService.findOrFail(id)
        return found.toDTO()
    }

    @DELETE
    @Path("/{id}")
    fun deleteById(@PathParam("id") id: UUID) {
        patientService.deleteById(id)
    }
}
